### ISSUE TITLE
Enable backup as root 2019

### ISSUE THREAD

**Issue by michaldybczak** [spersson/Kup#71]
Like in this thread:

https://github.com/spersson/Kup/issues/18

but I wanted to refresh the topic. Timeshift is a popular software nowadays, even implemented in Mint and recommended by many. However, Timeshift developer "reduced time developing it" meaning, he put a pause on its development. It's not abandonware but still, this is some concern and Kup already has all the structure to be a Timeshift KDE equivalent. All it needs is to have root support plus cli support.

I don't think that the arguments given in the old thread as relevant anymore. Kup is anyway a very obscure and not widely known software and one of the reasons is its lack of system backup capabilities. Now with timeshift not being actively developed, Kup would be an excellent choice for a software to take its place.

I'll forward this topic to Nate Graham and hopefully, he will start some official discussion about it, providing this is still an active project.

Kup is do damn close to being Timeshift replacement and it hurts to leave this in the same place as it is now (being not really competitive).

So please, reconsider this again. Old decisions didn't work out well, so there is no reason to try another route if KDE community is eager to accept this new way.

**Comment by Pointedstick** [comment]
I agree that it's really close to being a TimeShift replacement, and this is something I'd very much like as well.

FWIW I'm primarily a KDE developer, but I think it would be nice to get Kup incubated and become a KDE project: https://github.com/spersson/Kup/issues/76

Being a KDE project would bring greater visibility, more code contributions, someone else to handle releases and promo, and the possibility of greater integration into Plasma, up to and including being shipped by default as Plasma's official backup solution.

**Comment by michaldybczak** [comment]
Great! Now it depends on the developers if they are active and ready.

**Comment by spersson** [comment]
@michaldybczak 
Thanks for the feedback, always nice to hear someone interested in my software. You want me to reconsider a decision I made previously but you provide no reason other than "old decision didn't work out well". What makes you say that it hasn't worked well?

**Comment by michaldybczak** [comment]
Your reasoning you gave a few years ago for disabling root backup isn't holding up today.

Distros do want to integrate it (if I'm not mistaken Timeshift is already integrated in Mint, Manjaro considered doing the same but never came to realize that idea) and KDE might be interested in making Kup an official backup solution. Lack of system backup causes that the software is offering very basic options that many other programs also do have and thus being not competitive. Other apps providing already the same features are more known in the community. Most longtime Plasma users never heard about Kup and that says something about visibility if this program. I myself use grsync to backup home folders. Versioning isn't needed for home usage, but in high demand for system backup, at least for home usage. In a corporate environment, versioning is important for everything.
To stand out, it would great to offer full backup features because this is in high demand.

So in short, you gave up root support because as you said "Kup with this option would probably not be allowed to be shipped in most distros" but lack of root support is exactly the reason why it's not considered to be shipped with KDE-centric distros.

Timeshift is one of the best solutions of this time out there (there are others but timeshift became most popular) and became a standard app to advise for newbies because of its simplicity and power. Kup is like Qt version of timeshift, at least from UI perspective. However, as I mentioned earlier, timeshift is no longer in active development and not a Qt application so it can't be considered for an official KDE app. There is a void to fill.

Your program is very similar to timeshift and creates a confusing impression that it can do the same. There is no all around KDE backup app solution and Kup would fit nicely if it had those missing options:

- root support
- cli support
- optionally: Timeshift backup compatibility (this would ensure easy switching to Kup or between different DEs), this would be also a great marketing feature drawing attention

Since timeshift is opensource, you can even utilize its code to implement this. This would even have the benefit of the code being well tested (sure there are probably some fringe issues, no software is perfect). There is no need to reinvent the wheel.

Kup as official KDE project would benefit and had greater exposure. Win-win.

Probably there are some KDE requirements to be met first and Nate would be the person to talk about it.

**Comment by sorenstoutner** [comment]
I agree.  I would really like to use Kup to backup an entire system, including /etc and /var.

**Comment by Dzeri96** [comment]
Sorry for "bumping", but I'd really like this feature!
